File: Java/Estudo-API/API-Biblioteca-DevEficiente/src/main/java/io/github/monthalcantara/biblioteca/dto/response/LivroResponse.java
package io.github.monthalcantara.biblioteca.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import java.math.BigDecimal;

@Getter
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class LivroResponse {


    @JsonProperty("titulo")
    private String titulo;

    @JsonProperty("preco")
    private BigDecimal preco;

    @JsonProperty("isbn")
    private String isbn;

    @Deprecated
    private LivroResponse() {
    }

    @Override
    public String toString() {
        return "Livro {" +
                "titulo='" + titulo + '\'' +
                ", preco=" + preco +
                ", isbn='" + isbn + '\'' +
                '}';
    }
}

File: Kotlin/API-Casa_Do_Codigo_Kotlin/src/main/kotlin/io/github/monthalcantara/casadocodigo/dto/response/livro/LivroResponse.kt
package io.github.monthalcantara.casadocodigo.dto.response.livro

import io.github.monthalcantara.casadocodigo.dto.response.autor.AutorResponse
import io.github.monthalcantara.casadocodigo.dto.response.categoria.CategoriaResponse
import io.github.monthalcantara.casadocodigo.geraLinkHateoas
import io.github.monthalcantara.casadocodigo.model.Livro
import org.springframework.hateoas.Link
import org.springframework.hateoas.RepresentationModel
import java.math.BigDecimal

data class LivroResponse (
    val titulo: String,
    val resumo: String,
    val sumario: String,
    val preco: BigDecimal,
    val numeroPaginas: Int,
    val isbn: String,
    val categoria: CategoriaResponse,
    val autor: AutorResponse,
    val initialLink: Link
) : RepresentationModel<AutorResponse>(){
    constructor(livro: Livro) : this(
        titulo = livro.titulo,
        resumo = livro.resumo,
        sumario = livro.sumario,
        preco = livro.preco,
        numeroPaginas = livro.numeroPaginas,
        isbn = livro.isbn,
        categoria = CategoriaResponse.toResponse(livro.categoria),
        autor = AutorResponse.toResponse(livro.autor),initialLink = livro.id?.geraLinkHateoas()!!
        )

}
